File: Python/45. Jump Game II.py
class Solution:
    def jump(self, nums: list[int]) -> int:
        result = 0
        left = right = 0

        while right < len(nums)-1:
            farthest = 0
            for i in range(left, right+1):
                farthest = max(farthest, i+nums[i])
                print(farthest)
            left = right + 1
            right = farthest
            # print(left, right)
            result += 1

        return result

File: Javascript/45. Jump Game II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    let result = 0,
        left = 0,
        right = 0;
    
    while (right < nums.length-1) {
        let farthest = 0
        for (let i=left; i <= right; i++) {
            farthest = Math.max(farthest, i+nums[i])
        }
        left = right + 1
        right = farthest

        result++
    }

    return result
};
